runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -315,18 +315,73 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 			if (npcUtil.isDying(npc))
 			{
 				color = config.deadNpcMenuColor();
+				if (color != null)
+				{
+					final String target = ColorUtil.prependColorTag(Text.removeTags(event.getTarget()), color);
+					menuEntry.setTarget(target);
+					return;
+				}
 			}
 
-			if (color == null && highlightedNpcs.containsKey(npc) && config.highlightMenuNames() && (!npcUtil.isDying(npc) || !config.ignoreDeadNpcs()))
+			if (highlightedNpcs.containsKey(npc) && (!npcUtil.isDying(npc) || !config.ignoreDeadNpcs()))
 			{
 				color = MoreObjects.firstNonNull(getNpcHighlightColor(npc.getId()), config.highlightColor());
+				final String target = getTargetColor(event.getTarget(), config.highlightMenuStyle(), color);
+				menuEntry.setTarget(target);
 			}
+		}
+	}
+
+	private String getTargetColor(String target, NpcIndicatorsConfig.MenuHighlightStyle style, Color color)
+	{
+		if (style == NpcIndicatorsConfig.MenuHighlightStyle.NONE)
+		{
+			return target;
+		}
+
+		String cleanedTarget = Text.removeTags(target);
+		if (style == NpcIndicatorsConfig.MenuHighlightStyle.BOTH)
+		{
+			return ColorUtil.prependColorTag(cleanedTarget, color);
+		}
+
+		String name, level;
+		if (target.contains("  (level-"))
+		{
+			int levelStartIndex = cleanedTarget.lastIndexOf('(');
```

Comment:
this is necessary in the case that there is a level but there is no color tag on it. otherwise it will override the original 'name' color tag

Reply:
It isn't possible to have a level with no color tag

Reply:
oh well that simplifies it :) thanks

Reply:
though this PR does make it possible (if 'both' is selected)

Reply:
Just to be clear monster hp plugin will add an extra color tag to the level. This means if it's used with Npcindicators menu highlight, it will override the "remaining health" color part of the level. 

Arguably this is a bug, but I'm fine either way because it would work still, and at least the color is something chosen by the user.